wiki/Concept-Starting-and-Orchestrating-Bots.md

```diff
@@ -1,89 +0,0 @@
-
-# Communication with ark automate dektop app
-
-Multiple concepts could be applied in the case of ark_automate
-
-1. Polling and Long-Polling
-2. Websockets
-3. Server-sent events
-4. Higher Abstraction Frameworks
-
-### Polling and Long-Polling
-
-Polling is an approach where the desktop app would permenantly 'ask' that there is a new robot to execute. Although it is quite easy to implement, the performance is bad.
-Long Polling aims to solve the performance issue by just 'asking' one time and waiting for an answer. The desktop app would 'ask' the server again after a fixed periode of time if the server does not have any new robots to execute. In principle a great approach but it lacks support of frameworks one could use and implementing it from scratch would be too time consuming
-
-### Websockets
-
-When using websockets we would open a bidirectional communication channel between the server side and the client side. This is a great solution because real time communication is possible. Still, it is time consuming to implement and a successful connection is not always guaranteed as either element (proxy, firewall, …) prevent WebSocket connections between the client and the server or the browser does not support it yet.
-
-### Server-sent events
-
-This is an approach based on the concept of publish-subscribe. The client (desktop app) would subscribe to a 'channel' and the server would push new events (robots to be executed) on that channel. One of the main problem with SSE is the max of 6 client connections from a single host. Also, it is (subjectively) not the best approach to implement with node.js.
-
-### Higher Abstraction Frameworks
-
-There are some frameworks using one (or more) of these basic approaches but are one abstraction layer higher for the user. In particular, pusher and socket.io were looked at.
-Pusher is based on a publish subscribe approach simular to SSE and it seems to be very easy to use. The biggest disadvantage was that it is not open-source but ultimatly one would have to pay for it. Socket.io is a highly popular (> 4 Mio. weekly npm downloads) open source framework. It is mainly based on websocket communication but is also able to seamlessly switch to a long polling approach in the background if websockets do not work.
-
-### Decision
-
-In the end, we decided for socket.io as it is open-cource, well-supported, 'easy' to use, robust and websocket based enabling a bidirectional communication.
-To get started it is recommended reading [this](https://socket.io/docs/v4/index.html) introduction to socket.io. Especially these two subpages [1](https://socket.io/docs/v4/server-socket-instance/) & [2](https://socket.io/docs/v4/client-socket-instance/) are relevant for this usecase.
```

Comment:
I would leave that part on how we decided on the communication with the desktop app :)

Reply:
I think the decision for Socket.io and also the documentation of the implementation is important. I have added these two points directly again. However, I find the decision compared to the other communication methods not worth documenting or uninteresting for newcomers. What do you think about it @SanJSp ?

Reply:
I think the pages are irrelvant, if we want to keep something, then it would be this:

> In the end, we decided for socket.io as it is open-cource, well-supported, 'easy' to use, robust and websocket based enabling a bidirectional communication.

We could add it to the re-added socket.io description. See [this ](https://github.com/bptlab/ark_automate/pull/333#discussion_r642558813) comment

Reply:
Perfect Sandro! Exactly these lines I have already copied :D